crates/solver/src/settlement_submission/submitter.rs

```diff
@@ -474,16 +485,23 @@ impl<'a> Submitter<'a> {
             tx.clone().access_list(access_list.clone()).estimate_gas()
         )?;
 
+        ensure!(
+            gas_before_access_list > gas_after_access_list,
+            "access list exist but does not lower the gas usage"
+        );
+        ensure!(gas_before_access_list > U256::zero(), "bad gas estimation");
```

Comment:
This error will never happen because the previous line already ensures that `gas_before_access_list > 0`.

Reply:
Indeed. Nice.